src/signals/incident/components/IncidentNavigation/index.test.js

```diff
@@ -12,204 +15,124 @@ jest.mock('shared/services/auth/auth', () => ({
-      const withWizardWrapper = shallow(withWizard.get(0), { context });
+    expect(getByTestId('nextButton')).toBeInTheDocument();
+    expect(getByTestId('previousButton')).toBeInTheDocument();
+  });
 
-      withWizardWrapper.find('[data-testid="nextButton"]').simulate('click', event);
+  it('renders a previous button for the last step', () => {
+    const lastStep = [...steps].reverse()[0];
+
+    const { getByTestId } = render(
+      withAppContext(
+        <Wizard history={history}>
+          <Steps>
+            <Step id={lastStep}>
+              <IncidentNavigation {...props} />
+            </Step>
+          </Steps>
+        </Wizard>
+      )
+    );
 
-      expect(props.meta.handleSubmit).toHaveBeenCalledWith(event, context.wizard.next, undefined);
-    });
+    expect(getByTestId('previousButton')).toBeInTheDocument();
+  });
 
-    it('should trigger previous when clicking previous button', () => {
-      context.wizard.step = { id: 'incident/email' };
-      const withWizardWrapper = shallow(withWizard.get(0), { context });
+  it('does not render', () => {
+    const { queryByTestId } = render(
+      withAppContext(
+        <Wizard history={history}>
+          <Steps>
+            <Step id="incident/bedankt">
+              <IncidentNavigation {...props} />
+            </Step>
+          </Steps>
+        </Wizard>
+      )
+    );
 
-      withWizardWrapper.find('[data-testid="previousButton"]').simulate('click');
+    expect(queryByTestId('incident-navigation')).not.toBeInTheDocument();
+  });
 
-      expect(context.wizard.previous).toHaveBeenCalled();
-      expect(props.meta.handleSubmit).not.toHaveBeenCalled();
-    });
+  it('renders first step navigation for a step id that is not the correct format', () => {
+    const { queryByTestId } = render(
+      withAppContext(
+        <Wizard history={history}>
+          <Steps>
+            <Step>
+              <IncidentNavigation {...props} />
+            </Step>
+          </Steps>
+        </Wizard>
+      )
+    );
 
-    it('should trigger create when clicking submit button', () => {
-      getComponent();
+    expect(queryByTestId('incident-navigation')).not.toBeInTheDocument();
+  });
 
-      context.wizard.step = { id: 'incident/samenvatting' };
-      const withWizardWrapper = shallow(withWizard.get(0), { context });
+  it('should call onSubmit', () => {
+    const { getByTestId } = render(
+      withAppContext(
+        <Wizard history={history}>
+          <Steps>
+            <Step id={steps[0]}>
+              <IncidentNavigation {...props} />
+            </Step>
+          </Steps>
+        </Wizard>
+      )
+    );
 
-      withWizardWrapper.find('[data-testid="nextButton"]').simulate('click', event);
+    expect(handleSubmit).not.toHaveBeenCalled();
 
-      expect(props.meta.handleSubmit).toHaveBeenCalledWith(event, context.wizard.next, 'CREATE_INCIDENT');
+    act(() => {
+      fireEvent.click(getByTestId('nextButton'));
```

Comment:
You're missing a check that `handleSubmit` has been called after this statement.